### PR TITLE
[DNS] Adapter API changes and common calling handlers

### PR DESCRIPTION
**Pull request by prabhjot-msft**
# What
Adapter API changes and common calling handlers

# Why
https://skype.visualstudio.com/SPOOL/_workitems/edit/3811789
https://skype.visualstudio.com/SPOOL/_workitems/edit/3811792

# How Tested

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

**Comment by JamesBurnside** [comment]
> ## Calling bundle size is _**increased❗**_.
> * Current size: 9,414,815
> * Base size: 5,055,318
> * Diff size: 4,359,497

I don't think we can double our package size, these will need to be lazy loaded in some fashion

**Comment by mgamis-msft** [comment]
LGTM! Although the bundle size increase is quite a lot as James mentioned.

File: packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts [hunk 1]
```diff
@@ -895,20 +907,12 @@ export class AzureCommunicationCallAdapter<AgentType extends CallAgent | TeamsCa
 
   /* @conditional-compile-remove(DNS) */
   public async startNoiseSuppressionEffect(): Promise<void> {
-    const audioEffects: AudioEffectsStartConfig = {
-      noiseSuppression: new DeepNoiseSuppressionEffect()
-    };
-
-    await this.handlers.onStartNoiseSuppressionEffect(audioEffects);
+    await this.handlers.onStartNoiseSuppressionEffect();
```
Comment (mgamis-msft) [inline packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts:910]: Just curious. Why don't we expose the function arguments for our adapter function?

**Review by mgamis-msft — APPROVED** [review]

File: packages/communication-react/review/beta/communication-react.api.md [hunk 2]
```diff
@@ -3922,6 +3947,9 @@ export type OnRenderAvatarCallback = (
 userId?: string, options?: CustomAvatarOptions,
 defaultOnRender?: (props: CustomAvatarOptions) => JSX.Element) => JSX.Element | undefined;
 
+// @beta
+export const onResolveDNSDependencyLazy: () => Promise<DeepNoiseSuppressionEffectDependency>;
```
Comment (JamesBurnside) [inline packages/communication-react/review/beta/communication-react.api.md:3951]: Don't shorten here to DNS as DNS isn't used anywhere else, the name will be very long but `onResolveDeepNoiseSuppressionDependencyLazy`

File: packages/react-composites/src/composites/CallComposite/adapter/CallAdapter.ts [hunk 3]
```diff
@@ -168,6 +170,11 @@ export type CallAdapterClientState = {
    * Dependency to be injected for video background effect.
    */
   onResolveVideoEffectDependency?: () => Promise<VideoBackgroundEffectsDependency>;
+  /* @conditional-compile-remove(DNS) */
+  /**
+   * Dependency to be injected for deep noise suppression effect.
```
Comment (JamesBurnside) [inline packages/react-composites/src/composites/CallComposite/adapter/CallAdapter.ts:175]: mark as `@beta`

File: samples/Calling/src/app/views/CallScreen.tsx [hunk 4]
```diff
@@ -157,6 +159,10 @@ const AzureCommunicationCallScreen = (props: AzureCommunicationCallScreenProps):
         videoBackgroundImages,
         onResolveDependency: onResolveVideoEffectDependencyLazy
       },
+      /* @conditional-compile-remove(DNS) */
+      deepNoiseSuppressionOptions: {
+        onResolveDependency: onResolveDNSDependencyLazy
```
Comment (JamesBurnside) [inline samples/Calling/src/app/views/CallScreen.tsx:164]: Also add to callwithchat sample?

File: packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts (hunk 1 above)
Comment (prabhjot-msft) [reply packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts:910]: We are now providing the audioeffect through adapter option that is passed through the handlers, so we don't need to pass through adapters

File: packages/communication-react/review/beta/communication-react.api.md (hunk 2 above)
Comment (prabhjot-msft) [reply packages/communication-react/review/beta/communication-react.api.md]: done

File: packages/react-composites/src/composites/CallComposite/adapter/CallAdapter.ts (hunk 3 above)
Comment (prabhjot-msft) [reply packages/react-composites/src/composites/CallComposite/adapter/CallAdapter.ts:175]: done

File: samples/Calling/src/app/views/CallScreen.tsx (hunk 4 above)
Comment (prabhjot-msft) [reply samples/Calling/src/app/views/CallScreen.tsx]: done

**Review by JamesBurnside — APPROVED** [review]
Please fully account for the significant bundle size increases. This would be a significant regression if the primary bundle increases by 1+MB

**Pull request merged**